src/main/kotlin/org/rust/cargo/runconfig/Utils.kt

```diff
@@ -36,7 +36,16 @@ fun Project.buildProject() {
     val command = if (rustSettings.useCargoCheckForBuild) "check" else "build"
 
     for (cargoProject in cargoProjects.allProjects) {
-        val cmd = CargoCommandLine.forProject(cargoProject, command, listOf("--all"))
+        val allTargets = rustSettings.toolchain
+            ?.cargoOrWrapper(cargoProject.manifest.parent)
+            ?.supportsBuildCheckAllTargets ?: false
+        var arguments = listOf("--all")
+
+        if (allTargets) {
+            arguments += "--all-targets"
+        }
```

Comment:
It seems that we can create argument list one time outside of the loop

Reply:
Can we safely assume that the version of cargo given by `toolchain.rawCargo()` is the same as `cargoOrWrapper(cargoProject.manifest.parent)` for all `cargoProject`s?

Reply:
Why not? We use same cargo binary for all projects
Or do you know a case when we can't assume it?

Reply:
> Or do you know a case when we can't assume it?

I do not.  I did it this way just in case xargo could wrap around different cargo versions, but it doesn't seem to do anything like it (from a quick google search).